Apromore-Custom-Plugins/Log-Animation-Logic/src/main/java/org/apromore/service/loganimation/replay/BPMNDiagramHelper.java

```diff
@@ -95,150 +85,134 @@
-                            targetSet.add((FlowNode)sflow.getTargetRef());
-                        }
-                        targets.put((FlowNode)element, targetSet);
-                        
-                        sourceSet = new HashSet();
-                        for (SequenceFlow sflow : ((FlowNode)element).getIncomingSequenceFlows()) {
-                            sourceSet.add((FlowNode)sflow.getSourceRef());
-                        }
-                        sources.put((FlowNode)element, sourceSet);
-                        
-                        if (isDecision((FlowNode)element)) {
-                            decisions.add((FlowNode)element);
-                        }
-                        
-                        if (isMerge((FlowNode)element)) {
-                            merges.add((FlowNode)element);
-                        }
-                        
-                        if (isFork((FlowNode)element)) {
-                            forks.add((FlowNode)element);
-                        }
-                        
-                        if (isJoin((FlowNode)element)) {
-                            joins.add((FlowNode)element);
-                        }
-                        
-                        if (isORSplit((FlowNode)element)) {
-                            ORsplits.add((FlowNode)element);
-                        }
-                        
-                        if (isORJoin((FlowNode)element)) {
-                            ORjoins.add((FlowNode)element);
-                        }
-                        
-                        if (isActivity((FlowNode)element)) {
-                            activities.put(element.getName(), (FlowNode)element);
-                        }
-                        
-                        if (isMixedXOR((FlowNode)element)) {
-                            mixedXORs.add((FlowNode)element);
-                        }
-                        
-                        if (isMixedAND((FlowNode)element)) {
-                            mixedANDs.add((FlowNode)element);
-                        }
-                        
-                        if (isMixedOR((FlowNode)element)) {
-                            mixedORs.add((FlowNode)element);
-                        }
+                    for (SequenceFlow sflow : ((FlowNode) element).getOutgoingSequenceFlows()) {
+                        targetSet.add((FlowNode) sflow.getTargetRef());
+                    }
+                    targets.put((FlowNode) element, targetSet);
+
+                    sourceSet = new HashSet();
+                    for (SequenceFlow sflow : ((FlowNode) element).getIncomingSequenceFlows()) {
+                        sourceSet.add((FlowNode) sflow.getSourceRef());
+                    }
+                    sources.put((FlowNode) element, sourceSet);
+
+                    if (isDecision((FlowNode) element)) {
```

Comment:
Are these conditions mutually exclusive? If that's the case, perhaps switch would be better?